core/services/ocr2/plugins/generic/pipeline_runner_adapter.go

```diff
@@ -55,9 +55,8 @@ func (p *PipelineRunnerAdapter) ExecuteRun(ctx context.Context, spec string, var
 			ID:    trr.ID.String(),
 			Type:  string(trr.Task.Type()),
 			Index: int(trr.Task.OutputIndex()),
-
 			TaskValue: types.TaskValue{
-				Value:      trr.Result.Value,
+				Value:      trr.Result.OutputDB(),
```

Comment:
@george-dorin I'm probably missing something, but if we do this, then why do we need to move JSONSerializable into the GRPC layer? This was the workaround I was mentioning, so it feels like we need to do either this or the GRPC changes.